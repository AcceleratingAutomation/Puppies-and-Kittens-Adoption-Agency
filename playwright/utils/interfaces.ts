export interface LocalStorageData {
  displayName: string;
  token: string;
}

export interface LighthouseConfig {
  extends: string;
  settings: {
    formFactor?: "desktop" | "mobile";
    screenEmulation?: {
      mobile: boolean;
      width: number;
      height: number;
      deviceScaleFactor: number;
      disabled: boolean;
    };
  };
}
