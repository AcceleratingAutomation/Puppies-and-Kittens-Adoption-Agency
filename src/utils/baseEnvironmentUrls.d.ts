declare const envUrls: {
  local: {
    api: string;
    ui: string;
  };
  production: {
    api: string;
    ui: string;
  };
};
export = envUrls;
