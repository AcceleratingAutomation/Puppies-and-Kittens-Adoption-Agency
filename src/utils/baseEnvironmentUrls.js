module.exports = {
  local: {
    api: "http://localhost:5000",
    ui: "http://localhost:3000",
  },
  production: {
    api: "http://AcceleratingAutomation.com:5000",
    ui: "http://AcceleratingAutomation.com:3000",
  },
};
