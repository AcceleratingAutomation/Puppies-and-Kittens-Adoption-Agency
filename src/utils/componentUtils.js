import { fetchDetailsApi } from "../server/api/detailsApi";
import { deleteDetailsApi } from "../server/api/detailsApi";

export const fetchDetails = async (url, id, dispatch, setType, name) => {
    try {
      const data = await fetchDetailsApi(url, id);
      dispatch({
        type: setType,
        value: data[name],
      });
    } catch (err) {
      console.error(`Error fetching ${name} `, err.message);
    }
  };

export const deleteDetails = async (url, id, history, dispatch, path) => {
    try {
        if (await deleteDetailsApi(url, id)) {
            dispatch({ type: 'closeDialog' });
            history.push(path);
        }
    } catch (err) {
        console.error(`Error deleting ${id}`, err);
    }
};

export const navigateBack = (history, path) => {
    history.push(path);
};

export const navigateToEdit = (history, id, path) => {
    history.push(`${path}/${id}/edit`);
};