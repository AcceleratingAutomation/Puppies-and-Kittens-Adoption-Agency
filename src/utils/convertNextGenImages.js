const fs = require('fs');
const path = require('path');
const sharp = require('sharp');
const async = require('async');

function convertImage(input, output, callback, quality = 80) {
  sharp(input)
    .webp({ quality })
    .toFile(output, (err) => {
      if (err) {
        callback(new Error(`Error occurred: ${err}`));
      } else {
        callback(null);
      }
    });
}

function convertAllImagesInDirectory(directory) {
  fs.readdir(directory, (err, files) => {
    if (err) {
      throw new Error(`Error reading directory: ${err}`);
    }

    const imageFiles = files.filter((file) => ['.jpg', '.jpeg', '.png'].includes(path.extname(file)));

    // Convert 4 files in parallel to prevent overloading the CPU
    async.eachLimit(
      imageFiles,
      4,
      (file, callback) => {
        const inputPath = path.join(directory, file);
        const outputPath = path.join(
          directory,
          `${path.basename(file, path.extname(file))}.webp`,
        );
        convertImage(inputPath, outputPath, 80, callback);
      },
      (newErr) => {
        if (newErr) {
          throw new Error(`Error occurred during conversion: ${newErr}`);
        }
      },
    );
  });
}

// Only uncomment and run as needed to convert new images
convertAllImagesInDirectory('../../public/images/placeholders');

module.exports = {
  convertImage,
};
