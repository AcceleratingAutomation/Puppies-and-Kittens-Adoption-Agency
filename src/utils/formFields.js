const {
  breedLabel,
  confirmPasswordLabel,
  displayNameLabel,
  emailLabel,
  firstNameLabel,
  genderLabel,
  hasApplicationLabel,
  hasBackgroundCheckLabel,
  hasFosterLabel,
  hasVetLabel,
  imageLabel,
  isAcceptingLabel,
  isAdoptableLabel,
  isAdoptingLabel,
  isSterilizedLabel,
  isVaccinatedLabel,
  lastNameLabel,
  nameLabel,
  numCurrentRescuesLabel,
  numHouseholdPeopleLabel,
  numHouseholdPetsLabel,
  numTotalRescuesLabel,
  passwordLabel,
  roleLabel,
  typeLabel,
  userTypeLabel,
  usernameLabel,
} = require("../accessibility/accessibilityText");

// Rescue form fields
const rescueFormFields = [
  { id: "name", name: "name", type: "text", label: nameLabel },
  {
    id: "type",
    name: "type",
    type: "select",
    label: typeLabel,
    options: ["Cat", "Dog"],
  },
  {
    id: "gender",
    name: "gender",
    type: "select",
    label: genderLabel,
    options: ["Male", "Female"],
  },
  { id: "breed", name: "breed", type: "text", label: breedLabel },
  {
    id: "isSterilized",
    name: "isSterilized",
    type: "checkbox",
    label: isSterilizedLabel,
  },
  {
    id: "isVaccinated",
    name: "isVaccinated",
    type: "checkbox",
    label: isVaccinatedLabel,
  },
  {
    id: "isAdoptable",
    name: "isAdoptable",
    type: "checkbox",
    label: isAdoptableLabel,
  },
  {
    id: "hasFoster",
    name: "hasFoster",
    type: "checkbox",
    label: hasFosterLabel,
  },
  {
    id: "hasVet",
    name: "hasVet",
    type: "checkbox",
    label: hasVetLabel,
  },
  {
    id: "image",
    name: "image",
    type: "number",
    label: imageLabel,
  },
];

// Add user form fields
const addUserFormFields = [
  { id: "email", name: "email", type: "text", label: emailLabel },
  { id: "password", name: "password", type: "text", label: passwordLabel },
  {
    id: "confirmPassword",
    name: "confirmPassword",
    type: "text",
    label: confirmPasswordLabel,
  },
  { id: "username", name: "username", type: "text", label: usernameLabel },
  {
    id: "type",
    name: "type",
    type: "select",
    label: userTypeLabel,
    options: ["Adopter", "Foster", "Veterinarian", "Admin"],
  },
  {
    id: "role",
    name: "role",
    type: "select",
    label: roleLabel,
    options: ["adopter", "foster", "veterinarian", "admin"],
  },
];

// Common user form fields
const commonUserFormFields = [
  { id: "firstName", name: "firstName", type: "text", label: firstNameLabel },
  { id: "lastName", name: "lastName", type: "text", label: lastNameLabel },
  { id: "name", name: "name", type: "text", label: displayNameLabel },
  {
    id: "numHouseholdPeople",
    name: "numHouseholdPeople",
    type: "number",
    label: numHouseholdPeopleLabel,
  },
  {
    id: "numHouseholdPets",
    name: "numHouseholdPets",
    type: "number",
    label: numHouseholdPetsLabel,
  },
  {
    id: "hasBackgroundCheck",
    name: "hasBackgroundCheck",
    type: "checkbox",
    label: hasBackgroundCheckLabel,
  },
  {
    id: "image",
    name: "image",
    type: "number",
    label: imageLabel,
  },
];

// Unique user fields for numRescues
const numRescuesFormFields = [
  {
    id: "numCurrentRescues",
    name: "numCurrentRescues",
    type: "number",
    label: numCurrentRescuesLabel,
  },
  {
    id: "numTotalRescues",
    name: "numTotalRescues",
    type: "number",
    label: numTotalRescuesLabel,
  },
];

// Unique fields for Adopter Only
const adopterOnlyFormFields = [
  {
    id: "isAdopting",
    name: "isAdopting",
    type: "checkbox",
    label: isAdoptingLabel,
  },
  {
    id: "hasApplication",
    name: "hasApplication",
    type: "checkbox",
    label: hasApplicationLabel,
  },
];

// Unique user field for isAccepting
const isAcceptingFormFields = [
  {
    id: "isAccepting",
    name: "isAccepting",
    type: "checkbox",
    label: isAcceptingLabel,
  },
];

// Combine common fields with unique fields for each form
const adminFormFields = [...commonUserFormFields, ...numRescuesFormFields];
const adopterFormFields = [...commonUserFormFields, ...adopterOnlyFormFields];
const fosterFormFields = [
  ...commonUserFormFields,
  ...numRescuesFormFields,
  ...isAcceptingFormFields,
];
const veterinarianFormFields = [
  ...commonUserFormFields,
  ...numRescuesFormFields,
  ...isAcceptingFormFields,
];

export {
  addUserFormFields,
  adminFormFields,
  adopterFormFields,
  fosterFormFields,
  rescueFormFields,
  veterinarianFormFields,
};
