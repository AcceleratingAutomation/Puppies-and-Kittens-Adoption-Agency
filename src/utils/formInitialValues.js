const rescueInitialValues = {
  name: "",
  type: "",
  gender: "",
  breed: "",
  isSterilized: false,
  isVaccinated: false,
  isAdoptable: false,
  hasFoster: false,
  hasVet: false,
  bio: "",
  image: "",
};

const commonUserInitialValues = {
  firstName: "",
  lastName: "",
  name: "",
  numHouseholdPeople: 0,
  numHouseholdPets: 0,
  hasBackgroundCheck: false,
  bio: "",
  image: "",
};

const numRescuesInitialValues = {
  numCurrentRescues: 0,
  numTotalRescues: 0,
};

const isAcceptingRescuesInitialValues = {
  isAccepting: false,
};

const adoptersOnlyInitialValues = {
  isAdopting: false,
  hasApplication: false,
};

const adminInitialValues = {
  ...commonUserInitialValues,
  ...numRescuesInitialValues,
};
const adopterInitialValues = {
  ...commonUserInitialValues,
  ...adoptersOnlyInitialValues,
};
const fosterInitialValues = {
  ...commonUserInitialValues,
  ...numRescuesInitialValues,
  ...isAcceptingRescuesInitialValues,
};
const veterinarianInitialValues = {
  ...commonUserInitialValues,
  ...numRescuesInitialValues,
  ...isAcceptingRescuesInitialValues,
};

export {
  adminInitialValues,
  adopterInitialValues,
  fosterInitialValues,
  rescueInitialValues,
  veterinarianInitialValues,
};
