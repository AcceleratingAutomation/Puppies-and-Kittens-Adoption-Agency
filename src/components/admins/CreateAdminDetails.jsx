import React, { useCallback } from "react";
import { useHistory } from "react-router-dom";
import { Grid, Typography } from "@material-ui/core";
import { Formik } from "formik";
import { adminsUrl, adminsEndpoint } from "../../server/apiService/apiConfig";
import { AppHeader, tabs } from "../header/AppHeader";
import { createDetails, getTabValue } from "../../utils/componentUtils";
import { adminsText } from "../../accessibility/accessibilityText";
import { adminsValidationSchema } from "../../validations/usersValidationSchema";
import CreateAdminForm from "./CreateAdminForm";
import { adminInitialValues } from "../../utils/formInitialValues";

export default function CreateAdminDetails() {
  const history = useHistory();
  const tabValue = getTabValue(tabs, adminsText);

  const handleSubmit = useCallback(
    (values) => {
      createDetails(`${adminsUrl}/add`, values, history, adminsEndpoint);
    },
    [adminsUrl, history, adminsEndpoint],
  );

  return (
    <Formik
      initialValues={adminInitialValues}
      validationSchema={adminsValidationSchema}
      onSubmit={handleSubmit}
    >
      {({ errors, touched }) => (
        <div className="content">
          <AppHeader tabValue={tabValue} />
          <Grid
            container
            justify="center"
            alignItems="center"
            direction="column"
          >
            <Grid item style={{ marginBottom: "2vh" }}>
              <Typography variant="h3">Add New Admin</Typography>
            </Grid>
            <CreateAdminForm
              errors={errors}
              touched={touched}
              endpoint={adminsEndpoint}
            />
          </Grid>
        </div>
      )}
    </Formik>
  );
}
