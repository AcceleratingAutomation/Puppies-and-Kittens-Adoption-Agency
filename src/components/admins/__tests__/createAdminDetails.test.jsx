import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { getTabValue } from "../../../utils/componentUtils";
import { adminInitialValues } from "../../../utils/formInitialValues";
import CreateAdminDetails from "../CreateAdminDetails";
import {
  bioLabel,
  displayNameLabel,
  firstNameLabel,
  hasBackgroundCheckLabel,
  imageLabel,
  lastNameLabel,
  numCurrentRescuesLabel,
  numHouseholdPeopleLabel,
  numHouseholdPetsLabel,
  numTotalRescuesLabel,
  saveLabel,
  cancelLabel,
} from "../../../accessibility/accessibilityText";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

jest.mock("../../../utils/componentUtils", () => ({
  createDetails: jest.fn(),
  getTabValue: jest.fn(),
}));

describe("Create Admin Details", () => {
  beforeEach(() => {
    getTabValue.mockReturnValue(5);
    render(
      <Router>
        <CreateAdminDetails />
      </Router>,
    );
  });

  test("matches snapshot comparison", () => {
    const { asFragment } = render(
      <Router>
        <CreateAdminDetails />
      </Router>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("displays title", () => {
    expect(screen.getByText("Add New Admin")).toBeInTheDocument();
  });

  test("displays expected labels with form fields set to admin initial values", () => {
    expect(screen.getByLabelText(firstNameLabel).value).toBe(
      adminInitialValues.firstName,
    );
    expect(screen.getByLabelText(lastNameLabel).value).toBe(
      adminInitialValues.lastName,
    );
    expect(screen.getByLabelText(displayNameLabel).value).toBe(
      adminInitialValues.name,
    );
    expect(screen.getByLabelText(numHouseholdPeopleLabel).value).toBe(
      adminInitialValues.numHouseholdPeople.toString(),
    );
    expect(screen.getByLabelText(numHouseholdPetsLabel).value).toBe(
      adminInitialValues.numHouseholdPets.toString(),
    );
    expect(screen.getByLabelText(hasBackgroundCheckLabel).checked).toBe(
      adminInitialValues.hasBackgroundCheck,
    );
    expect(screen.getByLabelText(imageLabel).value).toBe(
      adminInitialValues.image,
    );
    expect(screen.getByLabelText(numCurrentRescuesLabel).value).toBe(
      adminInitialValues.numCurrentRescues.toString(),
    );
    expect(screen.getByLabelText(numTotalRescuesLabel).value).toBe(
      adminInitialValues.numTotalRescues.toString(),
    );
  });

  test("displays bio label", () => {
    expect(screen.getByText(bioLabel)).toBeInTheDocument();
  });

  test("displays expected buttons", () => {
    expect(screen.getByText(saveLabel)).toBeInTheDocument();
    expect(screen.getByText(cancelLabel)).toBeInTheDocument();
  });
});
