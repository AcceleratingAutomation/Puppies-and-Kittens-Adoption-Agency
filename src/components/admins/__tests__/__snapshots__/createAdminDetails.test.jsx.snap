// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Admin Details matches snapshot comparison 1`] = `
<DocumentFragment>
  <div
    class="content"
  >
    <header
      style="flex-grow: 1;"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <div
            class="PrivateHiddenCss-root PrivateHiddenCss-lgUp css-1d08qlc-PrivateHiddenCss-root"
          >
            <button
              aria-label="open drawer"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1d6w9lk-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="MenuIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="PrivateHiddenCss-root PrivateHiddenCss-mdDown css-cy70kq-PrivateHiddenCss-root"
          >
            <div
              class="MuiTabs-root css-1ujnqem-MuiTabs-root"
            >
              <div
                class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
                style="overflow: hidden; margin-bottom: 0px;"
              >
                <div
                  aria-label="Navigation tabs"
                  class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                  role="tablist"
                >
                  <button
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                    role="tab"
                    style="color: white;"
                    tabindex="-1"
                    type="button"
                  >
                    Favorites
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                    role="tab"
                    style="color: white;"
                    tabindex="-1"
                    type="button"
                  >
                    Rescues
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                    role="tab"
                    style="color: white;"
                    tabindex="-1"
                    type="button"
                  >
                    Adopters
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                    role="tab"
                    style="color: white;"
                    tabindex="-1"
                    type="button"
                  >
                    Fosters
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                    role="tab"
                    style="color: white;"
                    tabindex="-1"
                    type="button"
                  >
                    Veterinarians
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-selected="true"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                    role="tab"
                    style="color: white;"
                    tabindex="0"
                    type="button"
                  >
                    Admins
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <span
                  class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
                  style="left: 0px; width: 0px;"
                />
              </div>
            </div>
          </div>
          <div
            style="flex-grow: 1;"
          />
          <button
            aria-haspopup="true"
            aria-label="account of current user"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AccountCircleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6m0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
    </header>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1ewlecp-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        style="margin-bottom: 2vh;"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          Add New Admin
        </h3>
      </div>
      <form
        action="#"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container css-ql9ymi-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
              style="text-align: left;"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
              >
                <label
                  for="firstName"
                >
                  <strong>
                    First Name
                  </strong>
                  <input
                    id="firstName"
                    name="firstName"
                    style="margin: 0.625rem;"
                    type="text"
                    value=""
                  />
                </label>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
              >
                <label
                  for="lastName"
                >
                  <strong>
                    Last Name
                  </strong>
                  <input
                    id="lastName"
                    name="lastName"
                    style="margin: 0.625rem;"
                    type="text"
                    value=""
                  />
                </label>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
              >
                <label
                  for="name"
                >
                  <strong>
                    Display Name
                  </strong>
                  <input
                    id="name"
                    name="name"
                    style="margin: 0.625rem;"
                    type="text"
                    value=""
                  />
                </label>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
              >
                <label
                  for="numHouseholdPeople"
                >
                  <strong>
                    Number of People in Household
                  </strong>
                  <input
                    id="numHouseholdPeople"
                    name="numHouseholdPeople"
                    style="margin: 0.625rem;"
                    type="number"
                    value="0"
                  />
                </label>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
              >
                <label
                  for="numHouseholdPets"
                >
                  <strong>
                    Number of Pets in Household
                  </strong>
                  <input
                    id="numHouseholdPets"
                    name="numHouseholdPets"
                    style="margin: 0.625rem;"
                    type="number"
                    value="0"
                  />
                </label>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
              >
                <label
                  for="hasBackgroundCheck"
                >
                  <strong>
                    Has Background Check
                  </strong>
                  <input
                    id="hasBackgroundCheck"
                    name="hasBackgroundCheck"
                    style="margin: 0.625rem;"
                    type="checkbox"
                    value="false"
                  />
                </label>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
              >
                <label
                  for="image"
                >
                  <strong>
                    Image
                  </strong>
                  <input
                    id="image"
                    name="image"
                    style="margin: 0.625rem;"
                    type="number"
                    value=""
                  />
                </label>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
              >
                <label
                  for="numCurrentRescues"
                >
                  <strong>
                    Current Number of Rescues
                  </strong>
                  <input
                    id="numCurrentRescues"
                    name="numCurrentRescues"
                    style="margin: 0.625rem;"
                    type="number"
                    value="0"
                  />
                </label>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
              >
                <label
                  for="numTotalRescues"
                >
                  <strong>
                    Total Number of Rescues
                  </strong>
                  <input
                    id="numTotalRescues"
                    name="numTotalRescues"
                    style="margin: 0.625rem;"
                    type="number"
                    value="0"
                  />
                </label>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
              style="text-align: left;"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
              >
                <label
                  for="bio"
                >
                  <strong>
                    Bio
                  </strong>
                  <div
                    class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                    style="margin: 0.625rem; width: 100%;"
                  >
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <textarea
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                        id="bio"
                        name="bio"
                        rows="10"
                        style="height: 0px; overflow: hidden;"
                      />
                      <textarea
                        aria-hidden="true"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                        readonly=""
                        style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
                        tabindex="-1"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-ihdtdm"
                        >
                          <span
                            class="notranslate"
                          >
                            ​
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </label>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item css-1h16bbz-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary css-jh47zj-MuiButtonBase-root-MuiButton-root"
              style="margin: 0.625rem;"
              tabindex="0"
              type="submit"
            >
              Save
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary css-jh47zj-MuiButtonBase-root-MuiButton-root"
              style="margin: 0.625rem;"
              tabindex="0"
              type="button"
            >
              Cancel
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
